source/Geant/track/TrackState.hpp

```diff
@@ -225,7 +232,8 @@ GEANT_FORCE_INLINE
 void LinearStep(TrackState &state, double step)
 {
     state.fPhysicsState.fPstep -= step;
-    state.fStep += step;
+    state.fStep = step;
```

Comment:
What is the semantic of that change?  What does it fix?   I though 'fStep' was the distance travelled during that iteration of the step manager?  (So possibly multiple sub-step depending the distance to the next boundary and the physics length).   Is it possible that we are missing the reset to zero at the beginning of the step instead?

Reply:
Good point.  I think we need both current step and accumulated step, and only fStep was available.  So I added fTotalLength to accumulate, and fStep just keep current step, which means the change above is required.